mjordan/islandora_bagger
# Add support for JWT authentication

Issue #25 

This PR replaces all uses of HTTP basic authentication, which was not working in some cases, with the following flow:

Either 
- A token is passed in on the command-line, added by the Drupal module when in local mode, or
- The Generate Bag command makes a POST request to /user/login with the credentials in settings, and, thanks to the "Get JWT On Login" module, gets a token in the login response.
Then all subsequent HTTP requests pass the JWT in the headers.

### To test:

Run composer install to grab required updates.

Check out the related PR branch for islandora_drupal_integration.  Drush updatedb will print an error until https://drupal.org/project/getjwtonlogin is installed and enabled.

Existing workflows should be unchanged. No new settings are needed. 

However you can now generate bags from unpublished content as long as the user in the settings file has all the needed permissions.
